### DISCUSSION TITLE
Different movment selection

### DISCUSSION THREAD

**Discussion by N4tus** [helix-editor/helix#4863] [Ideas]
Using helix for my current project, the selection from movement commands are usually not what I want. Most of the time, when stepping with `w` through the words, there is too much white-space selected. What I really want is to jump from word to word and only select the current word, and have a way to extent the selection to include white-space when needed.

E.g. pressing `w` moves to the beginning of the next word and selects the current word (so the cursor is on the start and the word is selected to the end). Using `i` and `a` I now can edit at the beginning and end of the word. With this system moving with `e` is not really useful so I would replace it with an action that extends the selection to include the surrounding white-space and to remove if it is already selected.

**Comment by cem-al** [comment]
I agree, I am often using `*` to add the word to the register so I can jump to it's next/previous instance. `w` and `e` select the word plus white-space. Anyway to just select the next word without white-space?

**Reply by askreet** [reply]
You can use `_` to trim whitespace from selections, so the flow would be `w`, `_`, `*`.

**Comment by askreet** [comment]
I think the main motivation for it being the way it is is how `d` and `c` react with moving word-wise. If you selected only the whole words by default you would delete a word leaving two whitespaces in the middle of a sentence, for example. Adding a keybind that extends the selection to include surrounding whitespace creates the opposite problem, as you'd then have no whitespace left if you delete the selection.

The `_` keybind exists to trim the whitespace from your current selection. In my work, I tend to feel like that's the edge case versus being bothered by the additional selected whitespace.

**Comment by ellbur** [comment]
I can see why people like the whitespace-included behavior, but for me personally I would rather have the word-only behavior, at least as a command I could choose to use.

After deleting only a word, I can hit <kbd>d</kbd> again to delete the space, since the selection is now just the space. In my mind, that fits the design of helix (compared to Vim) of letting me operate incrementally and see the operation before I perform it.

**Reply by oderwat** [reply]
I have no hope that that will be changed but in my mind it would be "w" to select the word and "_" to add whitespace
